lib/taskana-core/src/test/java/acceptance/task/query/TaskQueryImplAccTest.java

```diff
@@ -2491,6 +2493,179 @@ void should_ApplyFilter_When_QueryingForNotWithin() {
+            .hasMessageContaining(
+                "The param \"withoutAttachment\" can only be used "
+                    + "without adding attributes of attachment as filter parameter");
+      }
+
+      @WithAccessId(user = "user-1-1")
+      @Test
+      void should_ThrowException_For_WithoutAttachmentAndAttachmentClassificationKeyLike() {
+        TaskQuery query =
+            taskService
+                .createTaskQuery()
+                .workbasketIdIn(wb.getId())
+                .withoutAttachment()
+                .attachmentClassificationKeyLike("%Key%");
+
+        assertThatThrownBy(() -> query.list())
+            .isInstanceOf(IllegalArgumentException.class)
+            .hasMessageContaining(
+                "The param \"withoutAttachment\" can only be used "
+                    + "without adding attributes of attachment as filter parameter");
+      }
+
+      @WithAccessId(user = "user-1-1")
+      @Test
+      void should_ThrowException_For_WithoutAttachmentAndAttachmentClassificationKeyNotIn() {
+        TaskQuery query =
+            taskService
+                .createTaskQuery()
+                .workbasketIdIn(wb.getId())
+                .withoutAttachment()
+                .attachmentClassificationKeyNotIn("Key");
+
+        assertThatThrownBy(() -> query.list())
+            .isInstanceOf(IllegalArgumentException.class)
+            .hasMessageContaining(
+                "The param \"withoutAttachment\" can only be used "
+                    + "without adding attributes of attachment as filter parameter");
+      }
+
+      @WithAccessId(user = "user-1-1")
+      @Test
+      void should_ThrowException_For_WithoutAttachmentAndAttachmentClassificationKeyNotLike() {
+        TaskQuery query =
+            taskService
+                .createTaskQuery()
+                .workbasketIdIn(wb.getId())
+                .withoutAttachment()
+                .attachmentClassificationKeyNotLike("Key");
+
+        assertThatThrownBy(() -> query.list())
+            .isInstanceOf(IllegalArgumentException.class)
+            .hasMessageContaining(
+                "The param \"withoutAttachment\" can only be used "
+                    + "without adding attributes of attachment as filter parameter");
+      }
+
+      @WithAccessId(user = "user-1-1")
+      @Test
+      void should_ThrowException_For_WithoutAttachmentAndAttachmentClassificationName() {
+        TaskQuery query =
+            taskService
+                .createTaskQuery()
+                .workbasketIdIn(wb.getId())
+                .withoutAttachment()
+                .attachmentClassificationNameIn("Name");
+
+        assertThatThrownBy(() -> query.list())
+            .isInstanceOf(IllegalArgumentException.class)
+            .hasMessageContaining(
+                "The param \"withoutAttachment\" can only be used "
+                    + "without adding attributes of attachment as filter parameter");
+      }
+    }
```

Comment:
Replace with `TestFactory`, so that all invalid combinations can be seen very fast.

Reply:
Ok :)